backend/src/main/java/saci/application/UserEvaluationController.java

```diff
@@ -0,0 +1,205 @@
+import jakarta.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import saci.domain.model.Knowledge;
+import saci.domain.model.Level;
+import saci.domain.model.Role;
+import saci.domain.service.KnowledgeService;
+import saci.domain.service.LevelService;
+import saci.domain.service.RoleService;
+
+@RestController
+@RequestMapping("/api/user-evaluation")
+@RequiredArgsConstructor
+@Validated
+public class UserEvaluationController {
+
+    private final RoleService roleService;
+    private final KnowledgeService knowledgeService;
+    private final LevelService levelService;
+
+    @Operation(summary = "Calculate the score for a role and return the current and next levels")
+    @ApiResponses(
+            value = {
+                @ApiResponse(
+                        responseCode = "200",
+                        description = "Score calculated successfully",
+                        content = {
+                            @Content(
+                                    mediaType = "application/json",
+                                    schema = @Schema(implementation = ScoreResponse.class))
+                        }),
+                @ApiResponse(
+                        responseCode = "404",
+                        description = "Role not found",
+                        content = @Content),
+                @ApiResponse(
+                        responseCode = "400",
+                        description = "Invalid request",
+                        content = @Content)
+            })
+    @PostMapping("/calculate-score")
+    public ResponseEntity<ScoreResponse> calculateScore(@Valid @RequestBody ScoreRequest request) {
+        Optional<Role> optionalRole = roleService.getRoleById(request.getRoleId());
+        if (!optionalRole.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        Role role = optionalRole.get();
+        List<Knowledge> knowledges = knowledgeService.getKnowledgesByRoleId(role.getId());
+
+        double totalWeightedScore = 0.0;
+        double totalWeight = 0.0;
+
+        for (Knowledge knowledge : knowledges) {
+            for (Answer answer : request.getAnswers()) {
+                if (knowledge.getId().equals(answer.getKnowledgeId())) {
+                    totalWeightedScore += answer.getAnswer() * knowledge.getWeight();
+                    totalWeight += knowledge.getWeight();
+                    break;
+                }
+            }
+        }
+
+        if (totalWeight == 0) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+
+        double score = (totalWeightedScore / (5 * totalWeight)) * 100;
```

Comment:
A good practice, is to assign random values to constants. This is a good practice because in this case, the value 5 could mean anything. If you assign 5 to a constant called MAX_WEIGHT, this value would have a meaning. Do this with the 100 as well (using the same casing, snake and upper case)